homeassistant/components/tuya/__init__.py

```diff
@@ -192,4 +149,27 @@ def async_remove_device(self, device_id: str) -> None:
         )
         if device_entry is not None:
             device_registry.async_remove_device(device_entry.id)
-            self.device_ids.discard(device_id)
+
+
+class TokenListener(SharingTokenListener):
+    """Token listener for upstream token updates."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        entry: ConfigEntry,
+    ) -> None:
+        """Init TokenListener."""
+        self.hass = hass
+        self.entry = entry
+
+    def update_token(self, token_info: str) -> None:
+        """Update token info in config entry."""
+        data = {**self.entry.data, "token_info": token_info}
```

Comment:
In the config flow we use the constant `CONF_TOKEN_INFO`, and the value for the key is a dict not a string.

Reply:
Hmm, well spotted, will look upstream on how they thought this would work...

Reply:
Turned out to be mostly a typing issue (not matching the parent class). I've adjusted the logic to be the same as the version used in the config entry.